src/scippneutron/file_loading/nxdata.py

```diff
@@ -79,16 +80,50 @@ def _getitem(self, select: ScippIndex) -> sc.DataArray:
                                                 index=index)
             signal.variances = sc.pow(stddevs, 2).values
         da = sc.DataArray(data=signal)
-        if self.attrs.get('axes', self._axes_default) is not None:
-            axes = self.attrs.get('axes', self._axes_default)
-            # Unlike self.dims we *drop* entries that are '.'
-            coords = [a for a in axes if a != '.']
-        else:
-            coords = self._signal.attrs['axes'].split(',')
-        for dim in coords:
-            index = to_plain_index([dim], select, ignore_missing=True)
-            da.coords[dim] = self._loader.load_dataset(self._group,
-                                                       dim,
-                                                       dimensions=[dim],
-                                                       index=index)
+
+        skip = [self._signal_name, self._errors_name]
+        skip += list(self.attrs.get('auxiliary_signals', []))
+        items = [k for k in self.keys() if isinstance(self[k], Field)]
+        items = [k for k in items if k not in skip]
+
+        for name in items:
+            # Newly written files should always contain indices attributes, but the
+            # standard recommends that readers should also make "best effort" guess
+            # since legacy files do not set this attribute.
+            indices = self.attrs.get(f'{name}_indices')
+            if indices is None:
```

Comment:
Do we want to start using this or does it hurt readability?
```suggestion
            if (indices := self.attrs.get(f'{name}_indices')) is None:
```

Reply:
I think it does hurt readability a bit. But probably it would just take time getting used to it.